src/main/java/com/google/googleidentity/oauth2/endpoint/AuthorizationEndpoint.java

```diff
@@ -162,27 +184,41 @@ private OAuth2Request parseOAuth2RequestFromHttpRequest(HttpServletRequest reque
                 OAuth2Utils.getUserSession(request).getUser().isPresent(),
                 "User should have logged in");
 
-        OAuth2Request.Builder oauth2RequestBuilder =
-                OAuth2Request.newBuilder()
-                        .setRequestAuth(
-                                OAuth2Request.RequestAuth.newBuilder()
-                                        .setClientId(client.getClientId())
-                                        .setUsername(
-                                                OAuth2Utils.getUserSession(request).getUser().get()
-                                                        .getUsername())
-                                        .build())
-                        .setRequestBody(
-                                OAuth2Request.RequestBody.newBuilder()
-                                        .setIsScoped(!scope.isEmpty())
-                                        .addAllScopes(scope)
-                                        .setResponseType(responseType)
-                                        .setRefreshable(
-                                                responseType.equals(
-                                                        OAuth2Constants.ResponseType.CODE))
-                                        .setGrantType(
-                                                OAuth2Utils.getGrantTypeFromResponseType(
-                                                        responseType))
-                                        .build());
+        OAuth2Request.Builder oauth2RequestBuilder = OAuth2Request.newBuilder();
+        oauth2RequestBuilder.getRequestAuthBuilder()
+                .setClientId(client.getClientId())
+                .setUsername(
+                        OAuth2Utils.getUserSession(request).getUser().get()
+                                .getUsername());
+        oauth2RequestBuilder.getRequestBodyBuilder()
+                .setIsScoped(!scope.isEmpty())
+                .addAllScopes(scope);
+        switch (responseType) {
+            case OAuth2Constants.ResponseType.CODE:
+                oauth2RequestBuilder.getRequestBodyBuilder().setResponseType(ResponseType.CODE);
+                break;
+            case OAuth2Constants.ResponseType.TOKEN:
+                oauth2RequestBuilder.getRequestBodyBuilder().setResponseType(ResponseType.TOKEN);
+            default:
+                // Will never happen since we have validated it
+                break;
+        }
```

Comment:
nit: do this with a map instead since that's basically what the switch statement is doing (unlike the grant_type switch statement below which also sets refreshable. That one is fine either as a map or switch IMO).

```
private static final ImmutableMap<String, ResponseType> RESPONSE_TYPE_MAP = ImmutableMap.builder()
    .put(OAuth2Constants.ResponseType.CODE, ResponseType.CODE)
    .put(OAuth2Constants.ResponseType.TOKEN, ResponseType.TOKEN)
    .build();

oauth2RequestBuilder.getRequestBodyBuilder()
    .setIsScoped(...)
    .addAllScopes(scope)
    .setResponseType(RESPONSE_TYPE_MAP.get(responseType));
```